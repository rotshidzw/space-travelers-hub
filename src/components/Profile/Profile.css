.profile-content {
  display: grid;
  grid-template-columns: 500px 500px;
  grid-template-rows: auto;
  column-gap: 20px;
  width: 100%;
  max-width: 1024px;
  margin-bottom: 60px;
  justify-content: space-between;
}

.mission-list {
  border: 1px solid goldenrod;
}

.rocket-list {
  border: 1px solid goldenrod;
}

.user-missions {
  list-style: none;
  padding: 0;
  margin: 0;
}

.user-rockets {
  list-style: none;
  padding: 0;
  margin: 0;
}

.user-rockets li,
.user-missions li {
  display: flex;
  border-bottom: 1px solid goldenrod;
  justify-content: space-between;
  padding: 20px;
  align-items: center;
}

.action-btns .leave-mission-btn,
.action-btns .cancel-rocket-btn {
  background-color: #f00;
  color: #fff;
  border: 1px solid #f00;
  padding: 5px;
  border-radius: 5px;
  cursor: pointer;
}

.action-btns .read-more-btn,
.action-btns .join-btn {
  background-color: transparent;
  border: 1px solid #6c757e;
  padding: 5px;
  border-radius: 5px;
  color: #6c757e;
  cursor: pointer;
}

.action-btns {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  flex: 1 1 40%;
  gap: 5px;
}

.mission-name {
  flex: 2 1 50%;
}
