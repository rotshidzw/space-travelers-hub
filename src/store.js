import { configureStore } from '@reduxjs/toolkit';
import RocketSlice from './Redux/Rocket/rocketSlice';
import MissionSlice from './Redux/Mission/missionSlice';

export default configureStore({
  reducer: {
    rockets: RocketSlice,
    missions: MissionSlice,

  },
//   middleware: customisedMiddleware,
});
