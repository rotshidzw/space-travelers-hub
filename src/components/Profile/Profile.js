/* eslint-disable react/prop-types */
import React from 'react';
import { useDispatch } from 'react-redux';
import { updateMission } from '../../Redux/Mission/missionSlice';
import { updateRocket } from '../../Redux/Rocket/rocketSlice';
import './Profile.css';

const Profile = (props) => {
  const { rockets, missions } = props;
  const dispatch = useDispatch();

  const handleMission = (event, id) => {
    dispatch(updateMission(id));
  };

  const handleReserversion = (event, id) => {
    dispatch(updateRocket(id));
  };

  const readMore = (event, url) => {
    window.open(url, '_blank', 'noopener,noreferrer');
  };

  return (
    <>
      <div className="profile-content">
        <div className="missions">
          <h1>Missions</h1>
          <div className="mission-list">
            <ul className="user-missions">
              {missions.length !== 0 ? (
                missions.map((x) => (
                  <li key={x.id}>
                    <p className="mission-name">{x.missionName}</p>
                    <div className="action-btns">
                      <button
                        onClick={(event) => handleMission(event, x.id)}
                        type="button"
                        className="leave-mission-btn"
                      >
                        Leave Mission
                      </button>
                      <button
                        onClick={(event) => readMore(event, x.url)}
                        type="button"
                        className="read-more-btn"
                      >
                        Read More
                      </button>
                    </div>
                  </li>
                ))
              ) : (
                <p>No Mission Joined</p>
              )}
            </ul>
          </div>
        </div>
        <div className="rockets">
          <h1 className="heading">Rockets</h1>
          <div className="rocket-list">
            <ul className="user-rockets">
              {rockets.length !== 0 ? (
                rockets.map((x) => (
                  <li key={x.id}>
                    <p className="rocket-name">{x.rocketName}</p>
                    <div className="action-btns">
                      <button onClick={(event) => handleReserversion(event, x.id)} type="button" className="cancel-rocket-btn">
                        Cancel Reservation
                      </button>
                      <button onClick={(event) => readMore(event, x.url)} type="button" className="read-more-btn">
                        Read More
                      </button>
                    </div>
                  </li>
                ))
              ) : (
                <p>No Rocket Reserved</p>
              )}
            </ul>
          </div>
        </div>
      </div>
    </>
  );
};

export default Profile;
