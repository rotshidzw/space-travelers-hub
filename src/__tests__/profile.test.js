import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../store';
import Profile from '../components/Profile/Profile';

it('Test if my_profile renderers correctly', () => {
  const missions = [];
  const rockets = [];
  const tree = renderer.create(
    <Provider store={store}>
      <Profile missions={missions} rockets={rockets} />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
