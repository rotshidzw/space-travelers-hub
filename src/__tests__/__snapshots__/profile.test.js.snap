// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if my_profile renderers correctly 1`] = `
<div
  className="profile-content"
>
  <div
    className="missions"
  >
    <h1>
      Missions
    </h1>
    <div
      className="mission-list"
    >
      <ul
        className="user-missions"
      >
        <p>
          No Mission Joined
        </p>
      </ul>
    </div>
  </div>
  <div
    className="rockets"
  >
    <h1
      className="heading"
    >
      Rockets
    </h1>
    <div
      className="rocket-list"
    >
      <ul
        className="user-rockets"
      >
        <p>
          No Rocket Reserved
        </p>
      </ul>
    </div>
  </div>
</div>
`;
